content/docs/components/switch/theming.mdx

```diff
@@ -2,3 +2,157 @@
 id: switch
 scope: theming
 ---
+
+The `Switch` component is a multipart component. The styling needs to be applied
+to each part specifically.
+
+> To learn more about styling multipart components, visit the
+> [Component Style](/docs/styled-system/component-style#styling-multipart-components)
+> page.
+
+## Anatomy
+
+- A: `container`
+- B: `thumb`
+- C: `track`
+
+## Theming properties
+
+The properties that affect the theming of the `Switch` component are:
+
+- `colorScheme`: The color scheme of the Switch component. Defaults to `blue`.
+- `size`: The size of the Switch component. Defaults to `md`.
+
+## Theming utilities
+
+- `createMultiStyleConfigHelpers`: a function that returns a set of utilities
+  for creating style configs for a multipart component (`definePartsStyle` and
+  `defineMultiStyleConfig`).
+- `definePartsStyle`: a function used to create multipart style objects.
+- `defineMultiStyleConfig`: a function used to define the style configuration
+  for a multipart component.
+
+```jsx live=false
+import { switchAnatomy } from '@chakra-ui/anatomy'
+import { createMultiStyleConfigHelpers } from '@chakra-ui/react'
+
+const { definePartsStyle, defineMultiStyleConfig } =
+  createMultiStyleConfigHelpers(switchAnatomy.keys)
+```
+
+## Customizing the default theme
+
+```jsx live=false
+import { switchAnatomy } from '@chakra-ui/anatomy'
+import { createMultiStyleConfigHelpers } from '@chakra-ui/react'
+
+const { definePartsStyle, defineMultiStyleConfig } =
+  createMultiStyleConfigHelpers(switchAnatomy.keys)
+
+const baseStyle = definePartsStyle({
+  // define the part you're going to style
+  container: {
+      // ...
+  },
+  thumb: {
+      bg: "red.500",
+  },
+  track: {
+      bg: "gray.100",
+      _checked: {
+      bg: "gray.100",
+      }
+  }
+})
+
+export const switchTheme = defineMultiStyleConfig({ baseStyle })
+```
+
+After customizing the switch theme, we can import it in our theme file and add
+it in the `components` property:
+
+```jsx live=false
+import { extendTheme } from '@chakra-ui/react'
+import { switchTheme } from './components/switch.ts'
+
+export const theme = extendTheme({
+  components: { Switch: switchTheme },
+})
+```
+
+> This is a crucial step to make sure that any changes that we make to the
+> switch theme are applied.
+
+
+## Using a custom color scheme
```

Comment:
You should put this block at the end of this section:

```markdown
Every time you're adding anything new to the theme, you'd need to run the CLI
command to get proper autocomplete in your IDE. You can learn more about the CLI
tool [here](/docs/styled-system/cli).
```

Reply:
Added it on https://github.com/chakra-ui/chakra-ui-docs/pull/1117/commits/e7d400ade4e8bd204d03648c977d8914a582dc9f